### PR TITLE
implemented MAD using scipy

### LINKED ISSUE #149: We should replace our own implementation of MAD by the one now available in scipy
https://github.com/sappelhoff/pyprep/blob/82390acff4c15a8e8ca92ed92c46f7a50d72e049/pyprep/utils.py#L464-L495

versus

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.median_abs_deviation.html

### PR DESCRIPTION
**Pull request by Ayush-Devs**
# PR Description:
this PR closes #149.
median_abs_deviation from scipy.stats is used to compute the MAD.

# Merge Checklist
- [ ] the PR has been reviewed and all comments are resolved
- [ ] all [CI][what-is-ci] checks pass
- [ ] (if applicable): the PR description includes the phrase `closes #<issue-number>` to [automatically close an issue][auto-close-documentation]
- [ ] (if applicable): bug fixes, new features, or [API][what-is-api] changes are documented in [whats_new.rst][whats-new-file]
- [ ] (if applicable): new contributors have added themselves to the authors list in the `CITATION.cff` file

### CODE REVIEW

**Review by sappelhoff — COMMENTED** [review]
Thanks for the changes, but the idea would be to replace our custom function `_mad` with the scipy implementation, instead of just calling the scipy function from within `_mad`. Would you be up for implementing this?

**Comment by Ayush-Devs** [comment]
just confirming before making the changes, you want me to remove the _mad function and directly implement scipy in the areas _mad function is called. Am I going in the right direction?

**Comment by Ayush-Devs** [comment]
The custom implementation of _mad raised a valueError when the input was not in 1D or 2D. Test_utils is expecting a valueerror which is not thrown by scipy.stats.median_abs_deviation. how really do you want me to fix this issue.

1) remove the test which expects the valueerror:
from test_utils.py line 166
    # Test exception with > 2-D arrays
    tst = np.random.rand(3, 3, 3)
    with pytest.raises(ValueError):
        median_abs_deviation(tst, axis=0)
        
2) create a custom function (i already did that in my first commit)

3) check the dimensions before the func is called.

i dont really understand the consequences to omit this check for 3d arrays. please tell me how should i proceed

File: pyprep/utils.py [hunk 1]
```diff
@@ -480,19 +481,14 @@ def _mad(x, axis=None):
         the MAD for each index along the specified axis.
 
     """
-    # Ensure array is either 1D or 2D
```
Comment (sappelhoff) [inline pyprep/utils.py:483]: please do not just delete comments if they are unrelated to the changes

**Review by sappelhoff — COMMENTED** [review]
I think we could also leave the changes as they are. But the current test suite needs to pass.

File: pyprep/utils.py (hunk 1 above)
Comment (Ayush-Devs) [reply pyprep/utils.py:483]: yes i will fix all these in my commit

**Comment by sappelhoff** [comment]
Great, thanks @Ayush-Devs could you please:

1. add a changelog entry
2. add yourself to the CITATION.cff file

Then we can merge this.

**Comment by Ayush-Devs** [comment]
I really hope I correctly updated the citation and changelog files. thank you so much for guiding me through this experience.

File: docs/whats_new.rst [hunk 2]
```diff
@@ -27,6 +27,7 @@ Version 0.5.0 (Unreleased)
 Changelog
 ~~~~~~~~~
 - :meth:`~pyprep.NoisyChannels.find_bad_by_nan_flat` now accepts a ``flat_threshold`` argument, by `Nabil Alibou`_ (:gh:`144`)
+- changed _mad function in utils.py to use median_abs_deviation from the sciPy module, by `Ayush Agarwal`_ (:gh:`153`).
```
Comment (sappelhoff) [inline docs/whats_new.rst:30]: looks good, except that you need to add a link to your name at the bottom of this file, see: `.. _Stefan Appelhoff: https://stefanappelhoff.com/` in the file --> just adjust this to fit for your name!

**Comment by sappelhoff** [comment]
Thanks @Ayush-Devs 🥳

**Pull request merged**